### PR TITLE
Improve options selections in messageActionSheet.

### PR DESCRIPTION
**Pull request by vishwesh3**
- Close drawer after selecting search or setting in StreamSidebar. 
- Close user drawer on selecting Search People.
- Navigate back on selecting narrow/reply from search screen. 
- Disable edit/delete option in special narrow.

### CODE REVIEW

File: src/nav/navigationSelectors.js [hunk 1]
```diff
@@ -11,3 +11,6 @@ export const getCurrentRouteParams = createSelector(
   getNavigationIndex,
   (routes, index) => routes[index].params,
 );
+
+export const getCurrentRouteName = (state: GlobalState): Object[] =>
```
Comment (borisyankov) [inline src/nav/navigationSelectors.js]: Let's just shorten it to `getCurrentRoute`. Also, the type of the returned value should be `string`.

Comment (vishwesh3) [inline src/nav/navigationSelectors.js]: Done 👍

File: src/conversations/ConversationsCard.js [hunk 2]
```diff
@@ -52,17 +52,23 @@ export default class ConversationsCard extends PureComponent {
...
+    const { conversations } = this.props;
 
     return (
       <View style={[componentStyles.container, styles.background]}>
         <ZulipButton
           secondary
           style={componentStyles.button}
           text="Search people"
-          onPress={actions.navigateToUsersScreen}
+          onPress={() => this.handleSearchPeople()}
```
Comment (vishwesh3) [inline src/conversations/ConversationsCard.js]: Yup! updated  👍

**Comment by vishwesh3** [comment]
updated :)

**Pull request merged**